src/generate_value_types.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+# 'register' is a deprecated keyword in C++1x
+gperf -t value_types_by_class.gperf | sed 's/register //g' > value_types_by_class.hpp
+gperf -t value_types_by_cql.gperf | sed 's/register //g' > value_types_by_cql.hpp
```

Comment:
I disagree with the no spaces in directory paths; Windows still loves spaces and a good number of those users still utilizes spaces in their paths especially when their user directory is their fullname which may contain spaces.

These are my concerns:

1. Using a shell script to generate these files requires that we also make a batch file for Windows; would prefer that we utilize CMake to handle the generation instead
2. Committing generated files could lead to a situation where we have forgotten to manually generate the files and commit; granted chances are low
3. Not making this an automated step during the build process turns this into a manual process when we should probably automate it

2) and 3) are kind of the same and I am not 100% against keeping this generation manual and committing the files when required as the amount of times these will change is extremely low.